### DISCUSSION TITLE
Issue with 'xvfb' Argument in Driver Initialization

### DISCUSSION THREAD

**Discussion by benfrankie** [seleniumbase/SeleniumBase#2949] [Q&A] [answered]
Hello Michael, 

I have encountered an issue when trying to initialize the Driver with the 'xvfb' argument. According to the documentation, I should be able to specify 'xvfb=True' directly in the Driver() constructor when running on Linux to manage the virtual framebuffer. However, doing so results in the following error:

Error during driver initialization: Driver() got an unexpected keyword argument 'xvfb'

Here's how I'm trying to initialize the Driver:
```
driver = Driver(
    uc=True,
    xvfb=True,
)
```
This issue occurs even though I have set up all necessary packages (xvfb, xvfbwrapper, etc.). Could you please clarify if there's a specific version or setup requirement that I might be missing, or if there's an alternative recommended approach to using xvfb with SeleniumBase?

Thank you

**Comment by mdmintz** [comment, marked as answer]
Use the `SB()` format for `xvfb`. The `Driver()` format does not have a context manager structure (there's no tearDown phase, only setUp), so it is limited in abilities.

**Reply by benfrankie** [reply]
I have been using SeleniumBase's undetectable mode (uc=True) for automation on a social media platform that utilizes Cloudflare and reCAPTCHA. Since the last major update to Cloudflare, I've observed a significant increase in the rate at which my accounts are being banned.

Could there be a correlation between the Cloudflare update and increased detectability of SeleniumBase’s UC mode? Have there been changes in the way Cloudflare detects automation tools that might affect the efficacy of uc_click, even with longer reconnect times (e.g., 30 seconds)?

Can switching to the SB() format inherently reduce the detectability of automated browser activities, even when visible CAPTCHAs are not present on the site being automated?

Does the `uc_gui_handle_cf` setting in SeleniumBase specifically target the bypassing of CAPTCHAs, or does it also contribute to a broader undetectability strategy by handling other elements that might flag automated behavior?

Any insights you could provide would be greatly appreciated.

**Reply by mdmintz** [reply]
Use the `SB(uc=True)` format with `sb.uc_gui_click_captcha()` for best results. Read all the docs: [UC Mode Documentation](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/uc_mode.md)
